src/schemas/m2cms.ts

```diff
@@ -0,0 +1,152 @@
+import process from 'process';
+import fs from 'node:fs';
+import path from 'node:path';
+import Typescript, {
+    type Node,
+    type InterfaceDeclaration,
+    type PropertySignature,
+    type Identifier
+} from 'typescript';
+import YAML from 'yaml';
+
+import Debug from 'debug';
+const DEBUG = Debug('m2cms')
+
+const KIND = (e: Node | undefined ) => e && Typescript.SyntaxKind[e.kind]
```

Comment:
> What does that return exactly? It's being inferred to string | undefined, but is that the intended signature?

it returns a Typescript Syntax node, or undefined.
this syntax node will describe the 'kind' of the token parsed.

Reply:
This isn't clear to me. How does the syntax node describe the kind?

Reply:
`Typescript.SyntaxKind[KIND]`

https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntaxkind?view=roslyn-dotnet-4.7.0

Reply:
these are all the valid 'kinds', we then filter using this `KIND macro` to handle only a few of those kinds.